worker/codalabworker/rest_client.py

```diff
@@ -138,22 +136,24 @@ def _upload_with_chunked_encoding(
                 to_send = fileobj.read(CHUNK_SIZE)
                 if not to_send:
                     break
-                conn.send('%X\r\n%s\r\n' % (len(to_send), to_send))
+                if type(to_send) is str:
+                    to_send = to_send.encode()
+                conn.send(b'%X\r\n%s\r\n' % (len(to_send), to_send))
```

Comment:
Previously (in Python 2) the code said:
```
conn.send('%X\r\n%s\r\n' % (len(to_send), to_send))
```

In that case, if `to_send` were a character such as `é` (which has unicode representation `\xc3\xa9`), then what would be sent in the connection would be the bytestring `1\r\n\xc3\xa9\r\n`.

However, the replacement Python 3 code I've added in would create the bytestring `2\r\n\xc3\xa9\r\n`. I'd think the new code I'm writing might be correct, because `é` is 2 bytes, not one byte, long -- but I don't know what the first number is used for later. Which is correct?

Reply:
@percyliang

Reply:
Resolved in #1336.